### PR TITLE
Edit saved searches

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Allow edition of saved searches.

To make it easier to notice that you are on a saved search, the "star" icon is now highlighted :
image

Clicking the "star" icon when you are already on a saved search allow you to update the search:
image

The 'Save as a new search" button allow you to copy the modified search to a new one instead of overriding the current one.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Internal tickets | !21016

### CODE REVIEW

File: ajax/savedsearch.php [hunk 1]
```diff
@@ -92,8 +93,23 @@
       $_GET['type']  =(int)$_GET['type'];
    }
 
+   $id = 0;
+   $saved_search = new SavedSearch();
+
+   // If an id was supplied in the query and that the matching saved searche
+   // is private OR the current used is allowed to edit public searches, then
```
Comment (trasher) [inline ajax/savedsearch.php:100]:
```suggestion
   // is private OR the current user is allowed to edit public searches, then
```

**Review by trasher — COMMENTED** [review]
I've not tested yet, sounds ok so far

File: ajax/savedsearch.php [hunk 2]
```diff
@@ -92,8 +93,23 @@
       $_GET['type']  =(int)$_GET['type'];
    }
 
+   $id = 0;
+   $saved_search = new SavedSearch();
+
+   // If an id was supplied in the query and that the matching saved searche
```
Comment (cedric-anne) [inline ajax/savedsearch.php:99]:
```suggestion
   // If an id was supplied in the query and that the matching saved search
```

File: ajax/savedsearch.php [hunk 3]
```diff
@@ -92,8 +93,23 @@
...
+   // is private OR the current user is allowed to edit public searches, then
+   // pass the id to showForm
+   if ($saved_search->getFromDB($_GET['id'] ?? 0)) {
```
Comment (cedric-anne) [inline ajax/savedsearch.php:102]: You could prevent a useless query.
```suggestion
   if (($requested_id = $_GET['id'] ?? 0) > 0 && $saved_search->getFromDB($requested_id)) {
```

**Review by cedric-anne — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**